### PR TITLE
Rework CHAdeMO support to more fully implement standard

### PR DESCRIPTION
**Pull request by smaresca**
**What**
This enhancement includes the appropriate CHAdeMO sequence for negotiation, connector locking and discharge support. Testing and further refinement is needed before this feature is entirely usable. Charging support is not yet present, though this PR lays the groundwork.

**Why**
To enable CHAdeMO vehicles to function as a battery, while retaining the safety afforded by a connector with locking solenoid, contactor control, and similar.

**How**
By implementing as much of the CHAdeMO standard as feasible given hardware limitation and access to closely-held specs.

### CODE REVIEW

File: Software/src/battery/CHADEMO-BATTERY.h [hunk 1]
```diff
@@ -5,6 +5,9 @@
 #include "../lib/miwagner-ESP32-Arduino-CAN/ESP32CAN.h"
 
 #define BATTERY_SELECTED
+#ifndef CONTACTOR_CONTROL
+#error Contactor control is required for CHADEMO support
```
Comment (smaresca) [inline Software/src/battery/CHADEMO-BATTERY.h:9]: This is causing a test to fail. Should the preprocessor #error be retained, hardcode the define for CONTACTOR_CONTROL, or some alternative?

Comment (dalathegreat) [reply Software/src/battery/CHADEMO-BATTERY.h:9]: Maybe code in a define for ContactorControl incase Chademo is selected? So users dont miss it?

File: Software/Software.ino [hunk 2]
```diff
@@ -453,6 +456,14 @@ void init_inverter() {
 void init_battery() {
   // Inform user what battery is used and perform setup
   setup_battery();
+
+#ifdef CHADEMO_BATTERY
+  intervalUpdateValues = 800;  // This mode requires the values to be updated faster
+#endif
+
+#ifndef BATTERY_SELECTED
```
Comment (dalathegreat) [inline Software/Software.ino:464]: This #ifndef BATTERY_SELECTED warning can be removed, it has been moved to the other file in another PR

File: Software/src/battery/CHADEMO-BATTERY.h (hunk 1 above)
Comment (smaresca) [reply Software/src/battery/CHADEMO-BATTERY.h:9]: I interpreted your comment as a suggestion to a #define CONTACTOR_CONTROL here in lieu of a test. Let me know if you meant something different

File: Software/Software.ino (hunk 2 above)
Comment (smaresca) [reply Software/Software.ino:464]: Removed

File: Software/src/battery/CHADEMO-BATTERY.h (hunk 1 above)
Comment (dalathegreat) [reply Software/src/battery/CHADEMO-BATTERY.h]: Yes, I meant that if a user has selected Chademo, the Chademo file would automatically also  define the CONTACTOR_CONTROL which is needed!

**Review by dalathegreat — APPROVED** [review]
